consensus/propagation/reactor.go

```diff
@@ -211,6 +211,11 @@ func (blockProp *Reactor) AddPeer(peer p2p.Peer) {
 		blockProp.Logger.Error("failed to get current compact block", "peer", peer.ID())
 		return
 	}
+	if len(cb.PartsHashes) == 0 {
```

Comment:
this avoids sending the compact block if it was created during catchup from a partset header.